web3/providers/websocket.py

```diff
@@ -0,0 +1,67 @@
+import asyncio
+import json
+import os
+from threading import (
+    Thread,
+)
+
+from web3.exceptions import (
+    NotConfigured,
+)
+from web3.providers.base import (
+    JSONBaseProvider,
+)
+
+try:
+    import websockets
+except ImportError:
+    websockets = None
+
+
+def _start_event_loop(loop):
+    asyncio.set_event_loop(loop)
+    loop.run_forever()
+
+
+# TODO: Think of a better name
+def _get_threaded_loop():
+    new_loop = asyncio.new_event_loop()
+    t = Thread(target=_start_event_loop, args=(new_loop,))
```

Comment:
*nitpick*

We've generally avoided single letter variables.  Can you rename this to something like `thread` or `loop_thread`

Reply:
will do